compiler/src/iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.td

```diff
@@ -520,26 +520,41 @@ def IREELinalgExt_AttentionOp : IREELinalgExt_Op<"attention",
        "getTiledImplementation"]>]> {
   let summary = "Attention operator";
   let description = [{
-    This operator takes in 3 tensors: query(Q), key(K) and value(V) and computes
-    the attention. For self-attention, all inputs have the same shape BxNxd where B is the
-    of the batch dimension, N is the sequence length and d is head dimension.
-    Typically N >>> d. Mathematically, the attention is defined as
-    matmul(softmax(matmul(Q, transpose(K))), V) and has shape BxNxd. Usually,
-    this operator also performs scaling, masking and dropout, but we leave
-    that out of the current implementation. For cross-attention, the query and output
-    have the same shape (BxNxd), while the key and value differ in sequence length
-    (they have shape BxLxd, where L != N).
-    This operator after tiling results in a tiled result as per flash attention and results
-    in the current `max` and `sum` statistics while processing the current tile.
+    Computes the scaled dot product attention function:
+
+    attention(Q, K, V, scale) = softmax(Q @ K.T * scale) @ V
+
+    Here Q, K, V are given tensors and scale is a scalar value specifying
+    the scale to use.
+
+    For self-attention, all inputs and the result have the same shape BxNxd
+    where B is the batch dimension, N is the sequence length and d is head
+    dimension. Typically N >>> d. Usually, this operator also performs
+    masking and dropout, but we leave that out of the current implementation.
+    For cross-attention, the query and output have the same shape (BxNxd),
+    while the key and value differ in sequence length (they have shape BxLxd,
+    where L != N).
+
+    This operator after tiling results in a tiled result as per
+    FlashAttention 2 and optionally results in the current `max` and `sum`
+    statistics while processing the current tile.
+
+    If transpose_v is speciifed, the V tensor passed as input is assumed to
+    be transposed:
+
+    attention(Q, K, V, scale) = softmax(Q @ K.T * scale) @ V.T
+
+    TODO: We should be moving to using a indexing map like approach so we
+    can generalize which tensor is transposed and which is not.
   }];
 
-  let arguments = (ins Variadic<AnyShaped>:$inputs,
+  let arguments = (ins Variadic<AnyType>:$inputs,
```

Comment:
One question - why not make the scale an attribute?

Reply:
Because the scale can be dynamically passed, example: dynamic head dimension

Reply:
+1 this should definitely be passed as an SSA value, not an attribute (also because that's how the torch op is defined).

Reply:
thanks